enhancements/sig-architecture/225-placement-score-visibility/README.md

```diff
@@ -0,0 +1,173 @@
+based on the scored state of clusters.
+
+In addition, it adds support for different sort strategies on the decisions generated by a Placement.
+By default, decisions will continue to be sorted by "Name". An additional sort strategy "Score" will
+enable sorting decisions numerically in descending order by prioritizer score.
+
+## Motivation
+
+Placement scores are a powerful feature of OCM for scheduling workloads, but there is currently
+no reliable way for users to see what scores were assigned to clusters for a placement. Scores are
+reported in events, but these are ephemeral. Since PlacementDecisions are orderd alphanumerically
+there is no good way to see the relative scores of clusters in a decision other than recomputing
+them for each cluster in the decision group(s).
+
+A user may wish to use Placement to find all clusters that match some conditions, but then pick
+only the top N clusters for their workload. While this could be accomplished by setting
+`numberOfClusters: N` on the placement, this means a new placement must be created for every value
+of N. Another use-case would be to distribute workloads out to selected clusters randomly weighted
+by their scores. There is currently no way to do this in OCM without recomputing scores for each cluster
+listed in the PlacementDecisions. Displaying scores in an easy to consume manner gives more flexibility
+to the user, and enables a single Placement to be reused for many different scheduling decisions
+(reducing API pressure on the hub).
+
+## Proposal
+
+Add `SortBy` to `PlacementSpec` to configure the sort order for PlacementDecision. This field
+is an enum with values "Name" and "Score". Add `Score` to `ClusterDecision` to reflect the
+latest score of each cluster in the decision.
+
+### Design Details
+
+#### API change
+
+Field for sorting decisions
+
+```go
+type PlacementSpec struct {
+  ...
+
+  // SortBy sets the sort order for decisions.
+  // It can be "Name", "Score", or "", where "" is "Name" by default.
+  // If sortBy is "Name", decisions will be orderd alphanumerically by cluster name
+  // If sortBy is "Score", decisions will be ordered numerically in descending order by score,
+  // then by cluster name in the event of a tie
+  // +optional
+  SortBy PlacementSortByType `json:"sortBy,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=Name;Score
+type PlacementSortByType string
+
+const (
+  // PlacementSortByName sorts decisions alphanumerically by cluster name
+  PlacementSortByName PlacementSortByType = "Name"
+  // PlacementSortByScore sorts decisions numerically in descending order by score.
+  // If one or more clusters are tied on score they will be sorted by name
+  PlacementSortByScore PlacementSortByType = "Score"
+)
+```
+
+Display score to consumer of decisions
+
+```go
+type ClusterDecision struct {
+  ...
+
+  // Score is the computed score for the cluster based on configured prioritizers
+  Score int64 `json:"score"`
```

Comment:
should this type be aligned with addonplacementscore https://github.com/open-cluster-management-io/api/blob/main/cluster/v1alpha1/types_addonplacementscore.go#L60 , int32.

Reply:
I used int64 because that is the type it is calculated in https://github.com/open-cluster-management-io/ocm/blob/4cbb12d5a2622ce5a698d5e07530f76166d3a922/pkg/placement/controllers/scheduling/schedule.go#L39-L40

Since final score is not bounded I think it makes sense to use int64